### PR TITLE
Fix macos interop toprod tests

### PR DESCRIPTION
**Pull request by jtattermusch**
and some extra cleanup to make that possible.

Fixes https://github.com/grpc/grpc/issues/19269

### CODE REVIEW

**Comment by jtattermusch** [comment]
Known failures:
https://github.com/grpc/grpc/issues/19271

**Comment by jtattermusch** [comment]
A successful run of macos interop_toprod tests https://source.cloud.google.com/results/invocations/16f66fce-331b-4788-bf2b-0b999d7d131c/targets/github%2Fgrpc%2Finterop_test/tests

File: tools/run_tests/run_interop_tests.py [hunk 1]
```diff
@@ -807,23 +807,17 @@ def compute_engine_creds_required(language, test_case):
     return False
 
 
-def auth_options(language,
-                 test_case,
-                 google_default_creds_use_key_file,
-                 service_account_key_file=None):
+def auth_options(language, test_case, google_default_creds_use_key_file,
```
Comment (apolcyn) [inline tools/run_tests/run_interop_tests.py]: small nit: the vertical parameters are slightly neater

Comment (jtattermusch) [reply tools/run_tests/run_interop_tests.py]: I think this was done by yapf

**Comment by jtattermusch** [comment]
We're using the new service account for all interop_to_prod tests (even on linux), but on linux we actually using the GCE default service account (obtained through VM metadata), so the defaults work well there.
The behavior of google_default_creds test with  --use_service_account_key file is a bit odd, because it sets the GOOGLE_APPLICATION_CREDENTIALS to a svc account key file, but still uses the default_service_account value - so on mac we need to override it for the test to work. On GCE linux VMs it works just fine.

**Pull request merged**